File: src/parser/__tests__/fixtures/go/sample1.go
// Package calculator provides basic arithmetic operations
package calculator

import (
	"errors"
	"fmt"
)

// Calculator represents a basic calculator
type Calculator struct {
	Memory float64
	History []Operation
}

// Operation represents a calculation operation
type Operation struct {
	Type string
	A    float64
	B    float64
	Result float64
}

// Adder is an interface for types that can add
type Adder interface {
	Add(a, b float64) float64
}

// Multiplier is an interface for multiplication
type Multiplier interface {
	Multiply(a, b float64) float64
}

// NewCalculator creates a new Calculator instance
func NewCalculator() *Calculator {
	return &Calculator{
		Memory: 0,
		History: make([]Operation, 0),
	}
}

// Add performs addition
func (c *Calculator) Add(a, b float64) float64 {
	result := a + b
	c.recordOperation("add", a, b, result)
	return result
}

// Subtract performs subtraction
func (c *Calculator) Subtract(a, b float64) float64 {
	result := a - b
	c.recordOperation("subtract", a, b, result)
	return result
}

// Multiply performs multiplication
func (c *Calculator) Multiply(a, b float64) float64 {
	result := a * b
	c.recordOperation("multiply", a, b, result)
	return result
}

// Divide performs division with error handling
func (c *Calculator) Divide(a, b float64) (float64, error) {
	if b == 0 {
		return 0, errors.New("division by zero")
	}
	result := a / b
	c.recordOperation("divide", a, b, result)
	return result, nil
}

// recordOperation is a private method to record operations
func (c *Calculator) recordOperation(opType string, a, b, result float64) {
	op := Operation{
		Type: opType,
		A: a,
		B: b,
		Result: result,
	}
	c.History = append(c.History, op)
}

// GetHistory returns the calculation history
func (c *Calculator) GetHistory() []Operation {
	return c.History
}

// ClearHistory clears the calculation history
func (c *Calculator) ClearHistory() {
	c.History = make([]Operation, 0)
}

// StoreToMemory stores a value in memory
func (c *Calculator) StoreToMemory(value float64) {
	c.Memory = value
}

// RecallFromMemory recalls the value from memory
func (c *Calculator) RecallFromMemory() float64 {
	return c.Memory
}

// CalculatorMode represents different calculator modes
type CalculatorMode int

const (
	StandardMode CalculatorMode = iota
	ScientificMode
	ProgrammerMode
)

// FormatResult formats a result for display
func FormatResult(value float64) string {
	return fmt.Sprintf("%.2f", value)
}

// Max returns the maximum of two numbers
func Max(a, b float64) float64 {
	if a > b {
		return a
	}
	return b
}

// Min returns the minimum of two numbers
func Min(a, b float64) float64 {
	if a < b {
		return a
	}
	return b
}

File: src/parser/__tests__/fixtures/java/sample1.java
// Calculator application demonstrating classes, interfaces, and methods
package com.example.calculator;

import java.util.ArrayList;
import java.util.List;

/**
 * Calculator interface for basic arithmetic operations
 */
public interface Calculator {
    double add(double a, double b);
    double subtract(double a, double b);
    double multiply(double a, double b);
    double divide(double a, double b) throws ArithmeticException;
}

/**
 * Operation represents a single calculation
 */
class Operation {
    private String type;
    private double operandA;
    private double operandB;
    private double result;

    public Operation(String type, double a, double b, double result) {
        this.type = type;
        this.operandA = a;
        this.operandB = b;
        this.result = result;
    }

    public String getType() {
        return type;
    }

    public double getResult() {
        return result;
    }
}

/**
 * BasicCalculator implements basic arithmetic operations
 */
public class BasicCalculator implements Calculator {
    private double memory;
    private List<Operation> history;

    public BasicCalculator() {
        this.memory = 0;
        this.history = new ArrayList<>();
    }

    @Override
    public double add(double a, double b) {
        double result = a + b;
        recordOperation("add", a, b, result);
        return result;
    }

    @Override
    public double subtract(double a, double b) {
        double result = a - b;
        recordOperation("subtract", a, b, result);
        return result;
    }

    @Override
    public double multiply(double a, double b) {
        double result = a * b;
        recordOperation("multiply", a, b, result);
        return result;
    }

    @Override
    public double divide(double a, double b) throws ArithmeticException {
        if (b == 0) {
            throw new ArithmeticException("Division by zero");
        }
        double result = a / b;
        recordOperation("divide", a, b, result);
        return result;
    }

    private void recordOperation(String type, double a, double b, double result) {
        history.add(new Operation(type, a, b, result));
    }

    public List<Operation> getHistory() {
        return new ArrayList<>(history);
    }

    public void clearHistory() {
        history.clear();
    }

    public void storeToMemory(double value) {
        this.memory = value;
    }

    public double recallFromMemory() {
        return this.memory;
    }

    public static double max(double a, double b) {
        return a > b ? a : b;
    }

    public static double min(double a, double b) {
        return a < b ? a : b;
    }
}

/**
 * ScientificCalculator extends BasicCalculator with advanced functions
 */
public class ScientificCalculator extends BasicCalculator {

    public double power(double base, double exponent) {
        return Math.pow(base, exponent);
    }

    public double sqrt(double value) {
        if (value < 0) {
            throw new ArithmeticException("Cannot take square root of negative number");
        }
        return Math.sqrt(value);
    }

    public double sin(double angle) {
        return Math.sin(angle);
    }

    public double cos(double angle) {
        return Math.cos(angle);
    }

    public double tan(double angle) {
        return Math.tan(angle);
    }
}

/**
 * CalculatorMode enum for different calculator modes
 */
enum CalculatorMode {
    STANDARD,
    SCIENTIFIC,
    PROGRAMMER
}

/**
 * CalculatorFactory for creating calculator instances
 */
class CalculatorFactory {
    public static Calculator createCalculator(CalculatorMode mode) {
        switch (mode) {
            case SCIENTIFIC:
                return new ScientificCalculator();
            case STANDARD:
            default:
                return new BasicCalculator();
        }
    }
}
